moveit_kinematics/test/test_kinematics_plugin.cpp

```diff
@@ -412,353 +390,358 @@ TEST_F(KinematicsTest, randomWalkIK)
+    moveit_msgs::msg::MoveItErrorCodes error_code;
+
+    RCLCPP_DEBUG(LOGGER, "Goal %f %f %f %f %f %f %f\n", goal.position.x, goal.position.y, goal.position.z,
+                 goal.orientation.x, goal.orientation.y, goal.orientation.z, goal.orientation.w);
+
+    kinematics_solver_->searchPositionIK(goal, seed_, timeout_,
+                                         const_cast<const std::vector<double>&>(consistency_limits_), sol, error_code);
+    ASSERT_EQ(error_code.val, error_code.SUCCESS);
+
+    // validate reached poses
+    std::vector<geometry_msgs::msg::Pose> reached_poses;
+    kinematics_solver_->getPositionFK(tip_frames, sol, reached_poses);
+    EXPECT_NEAR_POSES({ goal }, reached_poses, tolerance_);
+
+    // validate ground truth
+    if (!truth.empty())
+    {
+      ASSERT_EQ(truth.size(), sol.size()) << "Invalid size of ground truth joints vector";
+      Eigen::Map<Eigen::ArrayXd> solution(sol.data(), sol.size());
+      Eigen::Map<Eigen::ArrayXd> ground_truth(truth.data(), truth.size());
+      EXPECT_TRUE(solution.isApprox(ground_truth, 10 * tolerance_)) << solution.transpose() << std::endl
+                                                                    << ground_truth.transpose() << std::endl;
+    }
+  };
+
+  std::vector<double> ground_truth, pose_values;
+  constexpr char POSE_TYPE_RELATIVE[] = "relative";
+  constexpr char POSE_TYPE_ABSOLUTE[] = "absolute";
+
+  /* process tests definitions on parameter server of the form
+     pose_1:
+       pose:   [0.3, 0.2, 0.1, 0, 0, 0] // xyzrpy (position + euler angles)
+       joints: [0, 0, 0, 0, 0, 0]  // ground truth solution
+       type: relative  // pose applied relative to current pose
+     pose_2:
+       pose:   [0.1, 0.2, 0.3, 0, 0, 0, 0] // xyzwxyz (position + quaternion)
+       joints: [1, 2, 3, 4, 5, 6]  // pose applied absolute in planning frame
+       type: absolute
+  */
+  for (size_t i = 0; i < expected_test_poses; ++i)  // NOLINT(modernize-loop-convert)
+  {
+    const std::string pose_name = "pose_" + std::to_string(i);
+    const std::string pose_param = TEST_POSES_PARAM + "." + pose_name;
+    goal = initial;  // reset goal to initial
+    ground_truth.clear();
+
+    node_->get_parameter_or(pose_param + ".joints", ground_truth, ground_truth);
+    if (!ground_truth.empty())
+    {
+      ASSERT_EQ(ground_truth.size(), joints_.size())
+          << "Test pose '" << pose_name << "' has invalid 'joints' vector size";
+    }
+
+    pose_values.clear();
+    node_->get_parameter_or(pose_param + ".pose", pose_values, pose_values);
+    ASSERT_TRUE(pose_values.size() == 6 || pose_values.size() == 7)
+        << "Test pose '" << pose_name << "' has invalid 'pose' vector size";
+
+    Eigen::Isometry3d pose;
+    ASSERT_TRUE(parsePose(pose_values, pose)) << "Failed to parse 'pose' vector in: " << pose_name;
+    if (pose_values.size() == 7)
+    {
+      RCLCPP_DEBUG(LOGGER, "Pose Input (7): %f %f %f %f %f %f %f\n", pose_values.at(0), pose_values.at(1),
```

Comment:
I think the debug output can be removed

Reply:
Sure

Reply:
> I think this is still relevant, maybe you could use ASSERT_TRUE(node_->get_parameter(...)) without setting a default value. Then the tests would fail if a required parameter is not avaiilable

This makes sense I will add it.